### ISSUE TITLE
Metal Support

### ISSUE THREAD

**Issue by Dar-rius** [Dar-rius/zeroRL#59]
Adding Metal is simple, but it comes with a few bugs related to calculations.

We need to understand how to use the Metal backend without causing bugs or producing incorrect calculations; we’ll need to test it and monitor the quality of the training and the state of the gradients.

The resources to get started seem to be here [link](https://developer.apple.com/metal/pytorch/), but I encourage you to do more research.
